src/Ticket.java

```diff
@@ -1,121 +1,159 @@
-public class Ticket {
-    private String id;
-    private String concertHall;
-    private int eventCode;
+import annotation.NullableWarning;
+
+import java.util.Objects;
+
+public class Ticket extends IdentifiableEntity {
+    @NullableWarning
+    private final String concertHall;
+    private final int eventCode;
     private long time;
-    private boolean isPromo;
+    private final boolean isPromo;
+    @NullableWarning
     private String stadiumSector;
-    private double weight;
-    private double price;
+    private final double weight;
+    private final double price;
 
     private static final double MAX_WEIGHT = 1.000;
 
-    public Ticket(
-            String id,
-            String concertHall,
-            int eventCode,
-            boolean isPromo,
-            String stadiumSector,
-            double weight,
-            double price
-    ) {
-        validateAndInitialize(
-                id,
+
+    public Ticket(int id, String concertHall, int eventCode, boolean isPromo, String stadiumSector, double weight, double price) {
+        super(id);
+        validateFields(
                 concertHall,
                 eventCode,
-                isPromo,
                 stadiumSector,
-                weight,
-                price
+                weight
         );
+        this.concertHall = concertHall;
+        this.eventCode = eventCode;
+        this.time = System.currentTimeMillis() / 1000;
+        this.isPromo = isPromo;
+        this.stadiumSector = stadiumSector;
+        this.weight = weight;
+        this.price = price;
     }
 
-    public Ticket(
-            String concertHall,
-            int eventCode
-    ) {
-        validateAndInitialize(
-                null,
+    public Ticket(String concertHall, int eventCode) {
+        super(0);
```

Comment:
why 0? yiu can just provide in costructor, as you made line 20

Reply:
Ok, thank you)